src/neuroglancer/annotation/collection.ts

```diff
@@ -0,0 +1,309 @@
+ */
+
+import {Annotation, AnnotationReference, AnnotationSource, AnnotationType, Collection, LocalAnnotationSource} from 'neuroglancer/annotation';
+import {PlaceAnnotationTool, TwoStepAnnotationTool} from 'neuroglancer/annotation/annotation';
+import {PlaceBoundingBoxTool} from 'neuroglancer/annotation/bounding_box';
+import {PlaceSphereTool} from 'neuroglancer/annotation/ellipsoid';
+import {AnnotationLayerState} from 'neuroglancer/annotation/frontend';
+import {PlaceLineTool} from 'neuroglancer/annotation/line';
+import {PlaceLineStripTool} from 'neuroglancer/annotation/line_strip';
+import {PlacePointTool} from 'neuroglancer/annotation/point';
+import {PlaceSpokeTool} from 'neuroglancer/annotation/spoke';
+import {AnnotationRenderContext, AnnotationRenderHelper, registerAnnotationTypeRenderHandler} from 'neuroglancer/annotation/type_handler';
+import {MouseSelectionState} from 'neuroglancer/layer';
+import {StatusMessage} from 'neuroglancer/status';
+import {TrackableBoolean} from 'neuroglancer/trackable_boolean';
+import {UserLayerWithAnnotations} from 'neuroglancer/ui/annotations';
+import {registerTool} from 'neuroglancer/ui/tool';
+import {mat4, vec3} from 'neuroglancer/util/geom';
+
+const ANNOTATE_COLLECTION_TOOL_ID = 'annotateCollection';
+
+class RenderHelper extends AnnotationRenderHelper {
+  draw(context: AnnotationRenderContext) {
+    context;
+  }
+}
+
+registerAnnotationTypeRenderHandler(AnnotationType.COLLECTION, {
+  bytes: 3 * 4,
+  serializer: (buffer: ArrayBuffer, offset: number, numAnnotations: number) => {
+    const coordinates = new Float32Array(buffer, offset, numAnnotations * 3);
+    return (annotation: Collection, index: number) => {
+      const {source} = annotation;
+      const coordinateOffset = index * 3;
+      coordinates[coordinateOffset] = source[0];
+      coordinates[coordinateOffset + 1] = source[1];
+      coordinates[coordinateOffset + 2] = source[2];
+    };
+  },
+  sliceViewRenderHelper: RenderHelper,
+  perspectiveViewRenderHelper: RenderHelper,
+  pickIdsPerInstance: 1,
+  snapPosition: (position: vec3, objectToData, data, offset) => {
+    vec3.transformMat4(position, <vec3>new Float32Array(data, offset, 3), objectToData);
+  },
+  getRepresentativePoint: (objectToData, ann) => {
+    let repPoint = vec3.create();
+    vec3.transformMat4(repPoint, ann.source, objectToData);
+    return repPoint;
+  },
+  updateViaRepresentativePoint: (oldAnnotation, position: vec3, dataToObject: mat4) => {
+    let annotation = {...oldAnnotation};
+    annotation.source = vec3.transformMat4(vec3.create(), position, dataToObject);
+    // annotation.id = '';
+    return annotation;
+  }
+});
+
+export type MultiTool =
+    typeof PlacePointTool|typeof PlaceBoundingBoxTool|typeof PlaceLineTool|typeof PlaceSphereTool|
+    typeof PlaceLineStripTool|typeof PlaceSpokeTool|typeof MultiStepAnnotationTool;
+
+
+export class MultiStepAnnotationTool extends PlaceAnnotationTool {
```

Comment:
I think we should add an additional abstract class that extends the PlaceAnnotationTool that MultiStepAnnotationTool, LineStrip, and Spoke all inherit from. That way we can avoid having a class that has code that is never meant to be executed for instances of that class.

I also think we should add some access level modifiers to the methods of this class. E.g. updateLast and getChildRef are only called within the class so it should be private; appendNewChildAnnotation is called by children so it should be protected.